pilot/pkg/model/jwks_resolver.go

```diff
@@ -435,3 +435,41 @@ func (r *JwksResolver) refresh() {
 func (r *JwksResolver) Close() {
 	closeChan <- true
 }
+
+func compareJWKSResponse(oldKeyString string, newKeyString string) bool {
+	var oldJWKs map[string]interface{}
+	var newJWKs map[string]interface{}
+	e := json.Unmarshal([]byte(oldKeyString), &oldJWKs)
+	if e != nil {
+		log.Warnf("Previous JWKs public key JSON is not parseable: %s", oldKeyString)
+		return true
+	}
+	e = json.Unmarshal([]byte(newKeyString), &newJWKs)
+	if e != nil {
+		log.Warnf("New JWKs public key JSON is not parseable: %s", newKeyString)
+		return true
+	}
+
+	if oldJWKs == nil || newJWKs == nil {
+		return true
+	}
+
+	oldKeys := make(map[string]map[string]interface{})
```

Comment:
do we/should we have a struct we marshal this into instead of all of the `interface{}` casting?

Reply:
I started down that route and backed out since I don't think the format of the response is consistent for all providers. According to the spec there are specific required fields and then additional fields that could vary.

Reply:
Does it make sense to have a struct with just what we care about? You can configure the json unmarshal to ignore the unknown fields

Reply:
Can we sort the array "keys" using the "kid and then just compare with reflect.DeepEqual()?

Reply:
Thanks for that suggestion @yangminzhu, I modified the method to do that which makes it simpler. I also added in the original check up front that if the two strings are exactly equivalent we just return false which should often be the case from what I have seen.